packages/stratimikos/adapters/belos/src/Thyra_BelosLinearOpWithSolveFactory_def.hpp

```diff
@@ -691,193 +560,23 @@ void BelosLinearOpWithSolveFactory<Scalar>::initializeOpImpl(
-      }
-      break;
-    }
-    case SOLVER_TYPE_GCRODR:
-    {
-      // Set the PL
-      if(paramList_.get()) {
-        Teuchos::ParameterList &solverTypesPL = paramList_->sublist(SolverTypes_name);
-        Teuchos::ParameterList &gcrodrPL = solverTypesPL.sublist(GCRODR_name);
-        solverPL = Teuchos::rcp( &gcrodrPL, false );
-      }
-      // Create the solver
-      if (oldIterSolver != Teuchos::null) {
-        iterativeSolver = oldIterSolver;
-        iterativeSolver->setProblem( lp );
-        iterativeSolver->setParameters( solverPL );
-      } 
-      else {
-        iterativeSolver = rcp(new Belos::GCRODRSolMgr<Scalar,MV_t,LO_t>(lp,solverPL));
-      }
-      break;
-    }
-    case SOLVER_TYPE_RCG:
-    {
-      // Set the PL
-      if(paramList_.get()) {
-        Teuchos::ParameterList &solverTypesPL = paramList_->sublist(SolverTypes_name);
-        Teuchos::ParameterList &rcgPL = solverTypesPL.sublist(RCG_name);
-        solverPL = Teuchos::rcp( &rcgPL, false );
-      }
-      // Create the solver
-      if (oldIterSolver != Teuchos::null) {
-        iterativeSolver = oldIterSolver;
-        iterativeSolver->setProblem( lp );
-        iterativeSolver->setParameters( solverPL );
-      } 
-      else {
-        iterativeSolver = rcp(new Belos::RCGSolMgr<Scalar,MV_t,LO_t>(lp,solverPL));
-      }
-      break;
-    }
-    case SOLVER_TYPE_MINRES:
-    {
-      // Set the PL
-      if(paramList_.get()) {
-        Teuchos::ParameterList &solverTypesPL = paramList_->sublist(SolverTypes_name);
-        Teuchos::ParameterList &minresPL = solverTypesPL.sublist(MINRES_name);
-        solverPL = Teuchos::rcp( &minresPL, false );
-      }
-      // Create the solver
-      if (oldIterSolver != Teuchos::null) {
-        iterativeSolver = oldIterSolver;
-        iterativeSolver->setProblem( lp );
-        iterativeSolver->setParameters( solverPL );
-      }
-      else {
-        iterativeSolver = rcp(new Belos::MinresSolMgr<Scalar,MV_t,LO_t>(lp,solverPL));
-      }
-      break;
-    }
-    case SOLVER_TYPE_TFQMR:
-    {
-      // Set the PL
-      if(paramList_.get()) {
-        Teuchos::ParameterList &solverTypesPL = paramList_->sublist(SolverTypes_name);
-        Teuchos::ParameterList &minresPL = solverTypesPL.sublist(TFQMR_name);
-        solverPL = Teuchos::rcp( &minresPL, false );
-      }
-      // Create the solver
-      if (oldIterSolver != Teuchos::null) {
-        iterativeSolver = oldIterSolver;
-        iterativeSolver->setProblem( lp );
-        iterativeSolver->setParameters( solverPL );
-      }
-      else {
-        iterativeSolver = rcp(new Belos::TFQMRSolMgr<Scalar,MV_t,LO_t>(lp,solverPL));
-      }
-      break;
-    }
-
-    default:
-    {
-      TEUCHOS_TEST_FOR_EXCEPT(true);
-    }
+  Teuchos::ParameterList &solverTypesPL = paramList_->sublist(SolverTypes_name);
+  Teuchos::ParameterList &gmresPL = solverTypesPL.sublist(solverName_);
```

Comment:
Why is this called `gmresPL`?  I don't think it's specific to GMRES.

Reply:
Fixed.